taar/recommenders/collaborative_recommender.py

```diff
@@ -82,13 +82,15 @@ def recommend(self, client_data, limit):
         for addon in self.raw_item_matrix:
             # We don't really need to show the items we requested. They will always
             # end up with the greatest score.
-            if addon.get("id") in installed_addons:
+            hashed_id = str(addon.get("id"))
+            if hashed_id in installed_addons or\
+               hashed_id not in self.addon_mapping.keys():
```

Comment:
If `addon_mapping`is a dict, the `in` operator already looks for a key. You can omit `.keys()`. Also, I would wrap this condition in parenthesis and remove the backslash.